### PR TITLE
Go to direction when tapping list item

### LINKED ISSUE #519: Tapping on a maneuver in the directions list, should go to that maneuver on the map
This works in Open. It's pretty useful for previewing while navigating

### PR DESCRIPTION
**Pull request by sarahlensing**
- Migrate DirectionListAdapter to use ViewHolder pattern
- More Kotlin 1.0 syntax fixes

Closes #519

### CODE REVIEW

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionItemClickListener.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.mapzen.erasermap.view
+
+import com.mapzen.valhalla.Instruction
+
+/**
+ * Used by [DirectionListAdapter] when instantiating items
+ */
+interface DirectionItemClickListener {
+    abstract fun onDirectionItemClicked(position: Int)
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionItemClickListener.kt]: I'm not sure `abstract` is explicitly needed here. It should be implied in an interface.

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt [hunk 2]
```diff
@@ -6,8 +6,11 @@ import android.widget.ListView
 import com.mapzen.valhalla.Instruction
 import java.util.ArrayList
 
-public class DirectionListView(context: Context?, attrs: AttributeSet?) : ListView(context, attrs) {
-    public fun setInstructions(instructions: List<Instruction>) {
+class DirectionListView(context: Context?, attrs: AttributeSet?) : ListView(context, attrs) {
+
+    var directionItemClickListner: DirectionItemClickListener? = null
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt]: Spelling

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt [hunk 3]
```diff
@@ -19,14 +22,17 @@ public class DirectionListView(context: Context?, attrs: AttributeSet?) : ListVi
             distances.add(instruction.distance)
         }
 
-        adapter = DirectionListAdapter(context, strings, types, distances, false, false)
+        var directionAdapter = DirectionListAdapter(context, strings, types, distances, false, false)
```
Comment (ecgreb) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt]: Could this be `val`?

**Comment by ecgreb** [comment]
Logic looks good. Couple of style nits inline.

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionItemClickListener.kt (hunk 1 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionItemClickListener.kt]: You're right- rm'd it.

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt (hunk 2 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt]: Whoops, fixed :)

File: app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt (hunk 3 above)
Comment (sarahlensing) [inline app/src/main/kotlin/com/mapzen/erasermap/view/DirectionListView.kt]: Ah yes, changed to `val`

**Pull request merged**